@hts-forking/src/slotmap.js

```diff
@@ -0,0 +1,158 @@
+        this._map = new Map();
+    }
+
+    /**
+     *
+     * @param {bigint} slot
+     * @param {string} value
+     * @param {string} path
+     * @param {string} type
+     */
+    store(slot, value, path, type) {
+        const prev = this._map.get(slot);
+        if (prev !== undefined)
+            throw new Error(`Slot \`${slot}\` in use by ${JSON.stringify(prev)}: ${value}@${path}`);
+        this._map.set(slot, { value: '0x' + value, path, type });
+    }
+}
+
+/**
+ * @type {{[t_name: string]: (value: string) => string[]}}
+ */
+const _types = {
+    t_string_storage: function (str) {
+        return this['t_bytes_storage'](Buffer.from(str).toString('hex'));
+    },
+    t_bytes_storage: hexstr => {
+        const len = Buffer.from(hexstr, 'hex').length;
+        const [header, lenByte, chunks] =
+            len > 31
+                ? [
+                      '0',
+                      len * 2 + 1,
+                      [...Array(Math.ceil(len / 32)).keys()].map(i =>
+                          hexstr.substring(i * 64, (i + 1) * 64).padEnd(64, '0')
+                      ),
+                  ]
+                : [hexstr, len * 2, []];
+
+        return [`${header.padEnd(64 - 2, '0')}${lenByte.toString(16).padStart(2, '0')}`, ...chunks];
+    },
+    t_uint8: val => [toIntHex256(val)],
+    t_uint256: val => [toIntHex256(val)],
+    t_int256: str => [toIntHex256(str ?? 0)],
+    t_address: str => [toIntHex256(str?.replace('0.0.', '') ?? 0)],
+    t_bool: value => [toIntHex256(value ? 1 : 0)],
+};
+
+/**
+ * @param {Record<string, unknown>} token
+ * @returns {Map<bigint, {value: string, path: string}>}
+ */
+function slotMapOf(token) {
+    token['default_kyc_status'] = false;
+    token['treasury'] = token['treasury_account_id'];
+    token['ledger_id'] = '0x00';
+    token['second'] = `${token['expiry_timestamp']}`;
+    token['pause_status'] = token['pause_status'] === 'PAUSED';
+
+    token['token_keys'] = /**@type {const}*/ ([
+        ['admin_key', 0x1],
+        ['kyc_key', 0x2],
+        ['freeze_key', 0x4],
+        ['wipe_key', 0x8],
+        ['supply_key', 0x10],
+        ['fee_schedule_key', 0x20],
+        ['pause_key', 0x40],
+    ]).map(([prop, key_type]) => {
+        const key = token[prop];
+        if (key === null) return { key_type, ed25519: '', _e_c_d_s_a_secp256k1: '' };
+        assert(typeof key === 'object');
+        assert('_type' in key && 'key' in key);
+        if (key._type === 'ED25519')
+            return { key_type, ed25519: key.key, _e_c_d_s_a_secp256k1: '' };
+        return { key_type, ed25519: '', _e_c_d_s_a_secp256k1: key.key };
+    });
+
+    const customFees = /**@type {Record<string, unknown>}*/ (token['custom_fees']);
+    token['fixed_fees'] = customFees['fixed_fees'] ?? [];
+    token['fractional_fees'] = customFees['fractional_fees'] ?? [];
+    token['royalty_fees'] = customFees['royalty_fees'] ?? [];
```

Comment:
I don't think this would work, because the actual `TokenInfo` struct has very different fields for the custom fees from those which are in the mirror node response

Reply:
Yes, true that. Going to fix it.

Reply:
I think it would be better to tackle this in another PR. The reason being, our tests don't cover custom fees. This implies creating custom tokens with the desired fees, which in itself seems out of scope.

Reply:
Sure, sounds good. 

Btw, what I did to test things out for the custom fees (while doing the foundry solution) was to mock some data inside the `json` files for `USDC` and `MFCT` tokens and compare the results from `getTokenInfo` with the data

Reply:
Do you have any examples? Did you create the data manually? I would like to use the same test data for both Foundry and Hardhat.

Reply:
Yes, I found some real examples from random tokens on testnet and copied the custom fees from there

Reply:
Ok, at the end I managed to include some custom fee examples, so no need for a follow-up ticket for this.